### DISCUSSION TITLE
Consider making non-truncated asset filenames in Assets fields an accessibility or asset field setting

### DISCUSSION THREAD

**Discussion by mmikkel** [craftcms/cms#12236] [Ideas] [answered]
Since Craft 4.3.0, filenames in Assets fields are no longer visually truncated when the View Mode setting is set to "Large Thumbnails".

I appreciate that this change was likely done for accessibility reasons, but it's also proving to be a bit of a pain in some cases.

In my experience, a lot of the time the auto-generated asset titles are simply trash, and a lot of authors never bother to edit them into something actually useful.

Considering that the non-truncated titles mean that the asset titles now use significantly more screen real estate than before, and that the value in showing the full titles (vs. the more compact UI we had before) will probably vary a lot between projects (and clients/authors), I think it would be ideal if a "Show full asset titles in thumbnails" accessibility setting was introduced (or even an asset field setting, similar to the existing "Preview Mode" setting).

![image](https://user-images.githubusercontent.com/298510/201662650-89681f76-22be-4587-b30e-7bebd3ed1006.png)

**Comment by thupsi** [comment]
Seems like a legit concern. I would like to propose a small 'expand filename' toggle at the bottom of the individual asset UI that would be visible only when truncation occurs.

**Comment by brandonkelly** [comment, marked as answer]
We just released Craft 4.3.2 which rolls back the truncation change for large thumbnail titles. We have another fix in mind for the underlying accessibility issue, which we can implement in a future release.

**Reply by brandonkelly** [reply]
Asset title truncation is back in Craft 5.0.0-beta.6. Assets with truncated titles now show a tooltip on hover/focus with the full title (#14502).
